pdns/syncres.cc

```diff
@@ -2327,9 +2328,32 @@ uint32_t SyncRes::computeLowestTTD(const std::vector<DNSRecord>& records, const
 
     for(const auto& sig : signatures) {
       if (isRRSIGNotExpired(d_now.tv_sec, sig)) {
-        // we don't decerement d_sigexpire by 'now' because we actually want a TTD, not a TTL */
+        // we don't decrement d_sigexpire by 'now' because we actually want a TTD, not a TTL */
         lowestTTD = min(lowestTTD, static_cast<uint32_t>(sig->d_sigexpire));
       }
+      else {
+        /* if the signature IS expired, we should not cache this */
+        lowestTTD = d_now.tv_sec;
```

Comment:
This bit might be controversial, especially for backports. I think I should remove it, we already cap the TTL of Bogus records anyway.